src/pingdom-service-monitor.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/russellcardullo/go-pingdom/pingdom"
+	"log"
+	"net/url"
+	"strconv"
+)
+
+// PingdomService interfaces with monitor-proxy
+type PingdomService struct {
+	apiKey        string
+	username      string
+	password      string
+	url           string
+	alertContacts string
+}
+
+func (monitor *PingdomService) GetAll() []Monitor {
+	var monitors []Monitor
+	client := pingdom.NewClient(monitor.username, monitor.password, monitor.apiKey)
+	checks, err := client.Checks.List()
+	if err != nil {
+		log.Println("error recieved listing checks: " + err.Error())
+	}
+	for _, mon := range checks {
+		newMon := Monitor{
+			url:  mon.Hostname,
+			id:   fmt.Sprintf("%v", mon.ID),
+			name: mon.Name,
+		}
+		monitors = append(monitors, newMon)
+	}
+	return monitors
+}
+
+func (monitor *PingdomService) Add(m Monitor) {
+	client := pingdom.NewClient(monitor.username, monitor.password, monitor.apiKey)
+	checkConfig := m.setupPingdomConfigsFromAnnotations(m.annotations)
+	_, err := client.Checks.Create(&checkConfig)
+	if err != nil {
+		log.Println("error adding service:", err.Error())
+	}
+	log.Println("Added monitor for: ", m.name)
+}
+
+func (monitor *PingdomService) Update(m Monitor) {
+	client := pingdom.NewClient(monitor.username, monitor.password, monitor.apiKey)
```

Comment:
You're instantiating a new client everytime. Maybe create the client once in the setup method and reuse it?